### ISSUE TITLE
Chip Removable Icon to have button role

### ISSUE THREAD

**Issue by chase-skibeness** [primefaces/primereact#4190]
### Describe the feature you would like to see added

Currently it looks like a11y is forthcoming in Q2 2023 but I noticed something lacking in the spec of the chip component mentioned [here](https://primereact.org/chip/#:~:text=Accessibility-,Screen%20Reader,-Chip%20uses%20the)

The remove icon should have a button role property as clicking it performs a dom manipulation action in addition to the tab Index

### Is your feature request related to a problem?

I think this request addresses a potential gap PrimeReact might experience in the attempt to get WCAG ready

### Describe the solution you'd like

Right now I'm performing a useRef and traversing the dom to add the role manually but it would be nice if this element either came with this role property already or if it was possible to access the span that gets rendered via the API

**Comment by melloware** [comment]
I added this to 10.0.0 but note yeah all of hte buttons and links need to have their ARIA stuff revisited.
